apps/web/app/(dash)/dynamicisland.tsx

```diff
@@ -0,0 +1,275 @@
+    console.log(show);
+  }, [show]);
+
+  useEffect(() => {
+    document.addEventListener("keydown", (e) => {
+      if (e.key === "Escape") {
+        setshow(true);
+      }
+      console.log(e.key, lastBtn.current);
+      if (e.key === "a" && lastBtn.current === "Alt") {
+        setshow(false);
+      }
+      lastBtn.current = e.key;
+    });
+  }, []);
+  return (
+    <>
+      {show ? (
+        <div
+          onClick={() => setshow(!show)}
+          className="bg-[#1F2428] px-3 w-[2.23rem] overflow-hidden hover:w-[9.2rem] whitespace-nowrap py-2 rounded-3xl  transition-[width] cursor-pointer"
+        >
+          <div className="flex gap-4 items-center">
+            <Image src={AddIcon} alt="Add icon" />
+            Add Content
+          </div>
+        </div>
+      ) : (
+        <div>
+          <ToolBar cancelfn={cancelfn} />
+        </div>
+      )}
+    </>
+  );
+}
+
+const fakeitems = ["spaces", "page", "note"];
+
+function ToolBar({ cancelfn }: { cancelfn: () => void }) {
+  const [index, setIndex] = useState(0);
+  return (
+    <AnimatePresence mode="wait">
+      <motion.div
+        initial={{
+          opacity: 0,
+          y: 20,
+        }}
+        animate={{
+          y: 0,
+          opacity: 1,
+        }}
+        exit={{
+          opacity: 0,
+          y: 20,
+        }}
+        transition={{
+          duration: 0.2,
+        }}
+        className="flex flex-col items-center"
+      >
+        <div className="bg-[#1F2428] py-[.35rem] px-[.6rem] rounded-2xl">
+          <HoverEffect
+            items={fakeitems}
+            index={index}
+            indexFn={(i) => setIndex(i)}
+          />
+        </div>
+        {index === 0 ? (
+          <SpaceForm cancelfn={cancelfn} />
+        ) : index === 1 ? (
+          <PageForm cancelfn={cancelfn} />
+        ) : (
+          <NoteForm cancelfn={cancelfn} />
+        )}
+      </motion.div>
+    </AnimatePresence>
+  );
+}
+
+export const HoverEffect = ({
+  items,
+  index,
+  indexFn,
+}: {
+  items: string[];
+  index: number;
+  indexFn: (i: number) => void;
+}) => {
+  return (
+    <div className={"flex"}>
+      {items.map((item, idx) => (
+        <button
+          key={idx}
+          className="relative block h-full w-full px-2 py-1"
+          onClick={() => indexFn(idx)}
+        >
+          <AnimatePresence>
+            {index === idx && (
+              <motion.span
+                className="absolute inset-0 block h-full w-full rounded-xl bg-[#2B3237]"
+                layoutId="hoverBackground"
+                initial={{ opacity: 0 }}
+                animate={{
+                  opacity: 1,
+                  transition: { duration: 0.15 },
+                }}
+                exit={{
+                  opacity: 0,
+                  transition: { duration: 0.15, delay: 0.2 },
+                }}
+              />
+            )}
+          </AnimatePresence>
+          <h3 className="text-[#858B92] z-50 relative">{item}</h3>
```

Comment:
don't use any arbitrary values with #color like this, 
either add a colour to the theme, or use an existing one. This kinda becomes hard to maintain

Reply:
A lot of these changes are present with me locally , I will probably merge that later this day, I had been working locally for like 2 weeks without you knowing any changes, so I pushed that code just to make sure everyone gets to know the current state

Reply:
that's perfectly fine!